registry/storage/driver/middleware/alicdn/middleware.go

```diff
@@ -0,0 +1,116 @@
+package middleware
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	storagemiddleware "github.com/docker/distribution/registry/storage/driver/middleware"
+
+	"github.com/denverdino/aliyungo/cdn/auth"
+)
+
+// aliCdnStorageMiddleware provides a simple implementation of layerHandler that
+// constructs temporary signed AliCDN URLs from the storagedriver layer URL,
+// then issues HTTP Temporary Redirects to this AliCDN content URL.
+type aliCdnStorageMiddleware struct {
```

Comment:
Given that CDN is an acronym, I think this should be named `aliCDNStorageMiddleware` (see https://github.com/golang/go/wiki/CodeReviewComments#initialisms)